app/interactors/articles/destroy.rb

```diff
@@ -0,0 +1,17 @@
+module Articles
+  class Destroy
+    include Interactor
+
+    delegate :id, to: :context
+
+    def call
+      article = Article.find_by(id: id)
+
+      if article.nil?
+        context.fail!(errors: { base: ["Article not found"] })
+      elsif !article.destroy
+        context.fail!(errors: article.errors)
+      end
```

Comment:
Если сильно хочется 2 разных ошибки фейлить то так было бы красивее имхо

```suggestion
       context.fail!(errors: { base: ["Article not found"] }) if article.nil?
       context.fail!(errors: article.errors) unless article.destroy
```